### PR TITLE
Controllers, Repositories, Dto's

### PR DESCRIPTION
**Pull request by sedri14**

### CODE REVIEW

File: GetPet/GetPet.BusinessLogic/Model/AnimalTypeDto.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
```
Comment (Hadarlevi05) [inline GetPet/GetPet.BusinessLogic/Model/AnimalTypeDto.cs:1]: no need DTO for enum

File: GetPet/GetPet.BusinessLogic/Model/AnimalTypeDto.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace GetPet.BusinessLogic.Model
+{
+    public class AnimalTypeDto
+    {
+        public string Name { get; set; }
+    }
```
Comment (Hadarlevi05) [inline GetPet/GetPet.BusinessLogic/Model/AnimalTypeDto.cs:10]: add to all DTOs Id

File: GetPet/GetPet.BusinessLogic/Repositories/CityRepository.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        }
+
+        public new async Task DeleteAsync(int id)
+        {
+            await base.DeleteAsync(id);
+        }
+
+        public async Task<IEnumerable<City>> SearchAsync(BaseFilter filter)
+        {
+            var query = base.SearchAsync(entities.AsQueryable(), filter);
```
Comment (Hadarlevi05) [inline GetPet/GetPet.BusinessLogic/Repositories/CityRepository.cs:38]: support search city by name

File: GetPet/GetPet.BusinessLogic/Model/CityDto.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿
+namespace GetPet.BusinessLogic.Model
+{
+    public class CityDto
+    {
+        public string Name { get; set; }
```
Comment (Hadarlevi05) [inline GetPet/GetPet.BusinessLogic/Model/CityDto.cs:6]: add to all DTOs Id

File: GetPet/GetPet.BusinessLogic/Model/PetTraitDto.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace GetPet.BusinessLogic.Model
+{
+    public class PetTraitDto
+    {
+        public string PetName { get; set; }
```
Comment (Hadarlevi05) [inline GetPet/GetPet.BusinessLogic/Model/PetTraitDto.cs:9]: add to all DTOs Id

File: GetPet/GetPet.BusinessLogic/Model/TraitDto.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace GetPet.BusinessLogic.Model
+{
+    public class TraitDto
+    {
```
Comment (Hadarlevi05) [inline GetPet/GetPet.BusinessLogic/Model/TraitDto.cs:8]: add to all DTOs Id

File: GetPet/GetPet.BusinessLogic/Repositories/AnimalTypeRepository.cs [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        }
+
+        public new async Task DeleteAsync(int id)
+        {
+            await base.DeleteAsync(id);
+        }
+
+        public async Task<IEnumerable<AnimalType>> SearchAsync(BaseFilter filter)
+        {
+            var query = base.SearchAsync(entities.AsQueryable(), filter);
```
Comment (Hadarlevi05) [inline GetPet/GetPet.BusinessLogic/Repositories/AnimalTypeRepository.cs:37]: support search by name

File: GetPet/GetPet.BusinessLogic/Repositories/OrganizationRepository.cs [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        }
+
+        public new async Task DeleteAsync(int id)
+        {
+            await base.DeleteAsync(id);
+        }
+
+        public async Task<IEnumerable<Organization>> SearchAsync(BaseFilter filter)
+        {
+            var query = base.SearchAsync(entities.AsQueryable(), filter);
```
Comment (Hadarlevi05) [inline GetPet/GetPet.BusinessLogic/Repositories/OrganizationRepository.cs:37]: support search organization by name/Email

File: GetPet/GetPet.BusinessLogic/Repositories/PetTraitRepository.cs [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+        public new async Task DeleteAsync(int id)
+        {
+            await base.DeleteAsync(id);
+        }
+
+        public async Task<IEnumerable<PetTrait>> SearchAsync(BaseFilter filter)
+        {
+            var query = base.SearchAsync(entities.AsQueryable(), filter);
```
Comment (Hadarlevi05) [inline GetPet/GetPet.BusinessLogic/Repositories/PetTraitRepository.cs:39]: support search by trait name

File: GetPet/GetPet.BusinessLogic/Repositories/TraitRepository.cs [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+        }
+
+        public new async Task DeleteAsync(int id)
+        {
+            await base.DeleteAsync(id);
+        }
+
+        public async Task<IEnumerable<Trait>> SearchAsync(BaseFilter filter)
+        {
+            var query = base.SearchAsync(entities.AsQueryable(), filter);
```
Comment (Hadarlevi05) [inline GetPet/GetPet.BusinessLogic/Repositories/TraitRepository.cs:38]: support search by name

File: GetPet/GetPet.WebApi/Controllers/AnimalTypesController.cs [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+            IMapper mapper,
+            IAnimalTypeRepository animalTypeRepository)
+        {
+            _logger = logger;
+            _mapper = mapper;
+            _animalTypeRepository = animalTypeRepository;
+        }
+
+        [HttpGet]
+        public async Task<IEnumerable<AnimalTypeDto>> Get([FromQuery] BaseFilter filter)
```
Comment (Hadarlevi05) [inline GetPet/GetPet.WebApi/Controllers/AnimalTypesController.cs:30]: u need to return `Task<IActionResult>` and wrap the result in OK()
implement also post
check model validation

File: GetPet/GetPet.WebApi/Controllers/CitiesController.cs [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+                IMapper mapper,
+                ICityRepository cityRepository)
+        {
+            _logger = logger;
+            _mapper = mapper;
+            _cityRepository = cityRepository;
+        }
+
+        [HttpGet]
+        public async Task<IEnumerable<CityDto>> Get([FromQuery] BaseFilter filter)
```
Comment (Hadarlevi05) [inline GetPet/GetPet.WebApi/Controllers/CitiesController.cs:33]: u need to return Task<IActionResult> and wrap the result in OK()
implement also post
check model validation

File: GetPet/GetPet.WebApi/Controllers/OrganizationsController.cs [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+            IMapper mapper,
+            IOrganizationRepository organizationRepository)
+        {
+            _logger = logger;
+            _mapper = mapper;
+            _organizationRepository = organizationRepository;
+        }
+
+        [HttpGet]
+        public async Task<IEnumerable<OrganizationDto>> Get([FromQuery] BaseFilter filter)
```
Comment (Hadarlevi05) [inline GetPet/GetPet.WebApi/Controllers/OrganizationsController.cs:30]: u need to return Task<IActionResult> and wrap the result in OK()
implement also post
check model validation and throw a relevant error like 'Email already exists'..

File: GetPet/GetPet.WebApi/Controllers/PetTraitsController.cs [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            _logger = logger;
+            _mapper = mapper;
+            _petTraitRepository = petTraitRepository;
+        }
+
+        [HttpGet]
+        public async Task<IEnumerable<PetTraitDto>> Get([FromQuery] BaseFilter filter)
+        {
+            var petTraits = await _petTraitRepository.SearchAsync(filter);
```
Comment (Hadarlevi05) [inline GetPet/GetPet.WebApi/Controllers/PetTraitsController.cs:32]: u need to return Task<IActionResult> and wrap the result in OK()
implement also post
check model validation

File: GetPet/GetPet.WebApi/Controllers/TraitsController.cs [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+            IMapper mapper,
+            ITraitRepository traitRepository)
+        {
+            _logger = logger;
+            _mapper = mapper;
+            _traitRepository = traitRepository;
+        }
+
+        [HttpGet]
+        public async Task<IEnumerable<TraitDto>> Get([FromQuery] BaseFilter filter)
```
Comment (Hadarlevi05) [inline GetPet/GetPet.WebApi/Controllers/TraitsController.cs:31]: u need to return Task<IActionResult> and wrap the result in OK()
implement also post
check model validation

**Review by Hadarlevi05 — CHANGES_REQUESTED** [review]
pls review my comments

File: GetPet/GetPet.BusinessLogic/Repositories/PetTraitRepository.cs (hunk 9 above)
Comment (sedri14) [reply GetPet/GetPet.BusinessLogic/Repositories/PetTraitRepository.cs:39]: I didn't understand what should be returned.
all pets with this trait name?
all the possible values for this trait name..?
both?

**Review by Hadarlevi05 — APPROVED** [review]

**Pull request merged**